cmd/object-handlers.go

```diff
@@ -2106,7 +2106,15 @@ func (api objectAPIHandlers) PutObjectExtractHandler(w http.ResponseWriter, r *h
 		writeErrorResponse(ctx, w, toAPIError(ctx, err), r.URL)
 		return
 	}
-	metadata[xhttp.AmzStorageClass] = sc
+
+	// Remove the snowball header to avoid trying to untar files inside
+	// the archive when they are copied somewhere else later
+	delete(metadata, xhttp.AmzSnowballExtract)
```

Comment:
There are other headers as well such as specifying prefix, skipping dirs etc.

Reply:
Such as

- X-Amz-Meta-Minio-Snowball-Ignore-Dirs
- X-Amz-Meta-Minio-Snowball-Ignore-Errors
- X-Amz-Meta-Minio-Snowball-Prefix

Reply:
Like *all* `X-Amz-Meta-Minio-`? Are there any we want to keep?